packages/components/src/tabs/index.tsx

```diff
@@ -25,192 +25,206 @@ import { Tab } from './tab';
+			selectedTab?.dimmed,
+			setSelectedId,
+		] );
+
+		// Clear `selectedId` if the active tab is removed from the DOM in controlled mode.
+		useLayoutEffect( () => {
+			if ( ! isControlled ) {
+				return;
+			}
 
+			// Once the tabs have populated, if the `selectedTabId` still can't be
+			// found, clear the selection.
 			if (
-				! focusedElement ||
-				! items.some( ( item ) => focusedElement === item.element )
+				tabsHavePopulatedRef.current &&
+				!! selectedTabId &&
+				! selectedTab
 			) {
-				return; // Return early if no tabs are focused.
+				setSelectedId( null );
 			}
+		}, [ isControlled, selectedTab, selectedTabId, setSelectedId ] );
 
-			// If, after ariakit re-computes the active tab, that tab doesn't match
-			// the currently focused tab, then we force an update to ariakit to avoid
-			// any mismatches, especially when navigating to previous/next tab with
-			// arrow keys.
-			if ( activeId !== focusedElement.id ) {
-				setActiveId( focusedElement.id );
+		useEffect( () => {
+			// If there is no active tab, fallback to place focus on the first enabled tab
+			// so there is always an active element
+			if ( selectedTabId === null && ! activeId && firstEnabledTab?.id ) {
+				setActiveId( firstEnabledTab.id );
 			}
-		} );
-	}, [ activeId, isControlled, items, setActiveId ] );
+		}, [ selectedTabId, activeId, firstEnabledTab?.id, setActiveId ] );
 
-	const contextValue = useMemo(
-		() => ( {
-			store,
-			instanceId,
+		useEffect( () => {
+			if ( ! isControlled ) {
+				return;
+			}
+
+			requestAnimationFrame( () => {
+				const focusedElement =
+					items?.[ 0 ]?.element?.ownerDocument.activeElement;
+
+				if (
+					! focusedElement ||
+					! items.some( ( item ) => focusedElement === item.element )
+				) {
+					return; // Return early if no tabs are focused.
+				}
+
+				// If, after ariakit re-computes the active tab, that tab doesn't match
+				// the currently focused tab, then we force an update to ariakit to avoid
+				// any mismatches, especially when navigating to previous/next tab with
+				// arrow keys.
+				if ( activeId !== focusedElement.id ) {
+					setActiveId( focusedElement.id );
+				}
+			} );
+		}, [ activeId, isControlled, items, setActiveId ] );
+
+		const contextValue = useMemo(
+			() => ( {
+				store,
+				instanceId,
+			} ),
+			[ store, instanceId ]
+		);
+
+		return (
+			<TabsContext.Provider value={ contextValue }>
+				{ children }
+			</TabsContext.Provider>
+		);
+	},
+	{
+		Tab: Object.assign( Tab, {
+			displayName: 'Tabs.Tab',
 		} ),
-		[ store, instanceId ]
-	);
-
-	return (
-		<TabsContext.Provider value={ contextValue }>
-			{ children }
-		</TabsContext.Provider>
-	);
-}
-
-Tabs.TabList = TabList;
-Tabs.Tab = Tab;
-Tabs.TabPanel = TabPanel;
-Tabs.Context = TabsContext;
-
-export default Tabs;
+		TabList: Object.assign( TabList, {
+			displayName: 'Tabs.TabList',
+		} ),
+		TabPanel: Object.assign( TabPanel, {
+			displayName: 'Tabs.TabPanel',
+		} ),
```

Comment:
Did we also mean to expose `Tabs.Context`?

```suggestion
		} ),
		Context: Object.assign( TabsContext, {
			displayName: 'Tabs.Context',
		} ),
```

Probably yes, because there's existing usage:

https://github.com/WordPress/gutenberg/blob/37bceffa18cb6e6a89dcdfb68be79b81391f3266/packages/edit-widgets/src/components/sidebar/index.js#L94

https://github.com/WordPress/gutenberg/blob/f130e99ac69f4ce2cb9285363967975f96994e0f/packages/editor/src/components/sidebar/index.js#L61

Reply:
Yes, I meant to but forgot — thank you for catching it